src/CheckBox/CheckBox.tsx

```diff
@@ -1,64 +1,39 @@
+    label?: React.ReactNode;
+    /** Div wrapper props */
+    wrapperProps?: JSX.IntrinsicElements["div"];
+    /** Indicator for error, warning or success */
+    indicator?: Indicator;
+};
 
-const CheckBox: React.FunctionComponent<CheckBoxProps> = (props: CheckBoxProps): React.ReactElement<void> => {
+const CheckBox: React.FunctionComponent<CheckBoxProps> = ({ inline, description, label, wrapperProps, indicator, ...props }: CheckBoxProps): React.ReactElement<void> => {
     const [id, setId] = React.useState<string>(props.id);
-    const [formClass, setFormClass] = React.useState<string>("");
-
-    React.useEffect(() => {
-        setId(props.id ? id : randomId("checkbox-"));
-    }, [props.id]);
 
-    React.useEffect(() => {
-        let formClassName: string = "form-group custom-checkbox";
-        formClassName += props.inline ? " inline" : "";
-        formClassName += props.condensed ? " condensed" : "";
-        formClassName += props.className ? ` ${props.className}` : "";
-        setFormClass(formClassName);
-    }, [props.inline, props.condensed, props.className]);
+    React.useEffect(() => setId(props.id || label ? props.id || randomId("checkbox-") : null), [props.id, label]);
 
     return (
-        <div className={formClass}>
-            <div className="input-field">
-                {props.topLabel && (
-                    <label htmlFor={id} className="checkbox-toplabel">
-                        {props.topLabel}
-                    </label>
-                )}
-
-                <div className="custom-control">
-                    <input
-                        type="checkbox"
-                        className="custom-control-input"
-                        id={id}
-                        disabled={props.disabled}
-                        name={props.name}
-                        checked={props.checked}
-                        onChange={props.onChange}
-                        ref={props.reference}
-                    />
+        <div {...wrapperProps} className={classnames("seb", "checkbox", { inline }, wrapperProps?.className)}>
+            <div className={classnames("custom-control", "custom-checkbox", { "custom-control-inline": inline }, { [`is-${indicator?.type}`]: indicator })}>
+                <input {...props} type="checkbox" id={id} className={classnames("custom-control-input", props.className)} />
+                {label && (
                     <label className="custom-control-label" htmlFor={id}>
-                        {props.label}
-                        {props.description && <span className="checkbox-description">{props.description}</span>}
+                        <span className="checkbox-label">{label}</span>
                     </label>
-                </div>
+                )}
+                {description && <p className="checkbox-description">{description}</p>}
             </div>
+            {indicator && <FeedbackIndicator {...indicator} />}
```

Comment:
does this mean the feedback indicator is below the checkbox? Shouldn't the indicator be a wrapper around the checkbox?

Reply:
it's a wrapper when you need it, inside each checkbox, it's only a message, the border is handled by the checkbox component itself